src/devices/bus/mc10/mc10_mcx128.cpp

```diff
@@ -0,0 +1,379 @@
+
+    mc10_mcx128.cpp
+
+    Code for emulating Darren Atkinson's MCX-128 cartridge
+
+	Features:
+		128K of RAM expansion
+		16K of ROM expansion
+
+	New MC-10 memory map defined by this cartridge:
+
+	0000 - 0003	6803 Ports
+	0004 - 0007	Expansion RAM
+	0008 - 000E	6803 Status / Control
+	000F		Expansion RAM
+	0010 - 0013	6803 UART
+	0014		6803 RAM Control Reg
+	0015 - 001F	Unused
+	0020 - 007F	Expansion RAM
+	0080 - 00FF	On-chip CPU RAM / Expansion RAM
+	0100 - 3FFF	Expansion RAM
+	4000 - 5FFF	Built-In RAM / Expansion RAM
+	6000 - BEFF	Expansion RAM
+	BF00		RAM Bank Control Reg
+	BF01		ROM Map Control Reg
+	BF80 - BFFF	Keyboard / VDG / Sound
+	C000 - DFFF	EPROM or Expansion RAM
+	E000 - FFFF	Built-in ROM, EPROM or Expansion RAM
+
+	Deficiency:
+
+	* Writing to 6803 chip RAM at $80 to $ff should be mirrored
+	  to external RAM.
+
+***************************************************************************/
+
+#include "emu.h"
+#include "mc10_mcx128.h"
+
+// #include "machine/ram.h"
+
+//#define VERBOSE (LOG_GENERAL )
+#include "logmacro.h"
+
+
+ROM_START(mc10_mcx128)
+	ROM_REGION(0x4000, "eprom", ROMREGION_ERASE00)
+	ROM_LOAD("mcx128bas.rom", 0x0000, 0x4000, CRC(11202e4b) SHA1(36c30d0f198a1bffee88ef29d92f2401447a91f4))
+ROM_END
+
+ROM_START(alice_mcx128)
+	ROM_REGION(0x4000, "eprom", ROMREGION_ERASE00)
+	ROM_LOAD("alice128bas.rom", 0x0000, 0x4000, CRC(a737544a) SHA1(c8fd92705fc42deb6a0ffac6274e27fd61ecd4cc))
+ROM_END
+
+//**************************************************************************
+//  TYPE DECLARATIONS
+//**************************************************************************
+
+// ======================> mc10_pak_device
+
+class mc10_pak_mcx128_device :
+		public device_t,
+		public device_mc10cart_interface
+{
+public:
+	// construction/destruction
+	mc10_pak_mcx128_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+	mc10_pak_mcx128_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock);
+
+protected:
+	// device-level overrides
+	virtual void device_start() override;
+	virtual void device_reset() override;
+	virtual void device_post_load() override;
+
+	virtual const tiny_rom_entry *device_rom_region() const override
+	{
+		return ROM_NAME(mc10_mcx128);
+	}
+
+	u8 control_register_read(offs_t offset);
+	void control_register_write(offs_t offset, u8 data);
+
+	void write_ram_mirror(address_space &space, offs_t offset, u8 data);
+
+	void view_map0(address_map &map);
+	void view_map1(address_map &map);
+	void view_map2(address_map &map);
+	void view_map3(address_map &map);
+	void view_map4(address_map &map);
+	void view_map5(address_map &map);
+	void view_map6(address_map &map);
+	void view_map7(address_map &map);
+
+
+private:
+	memory_share_creator<u8> m_share;
+	memory_view m_view;
+	memory_bank_creator m_bank0, m_bank1, m_bank2, m_bank3, m_bank4, m_bank5, m_bank6, m_bank7;
```

Comment:
Candidate for `memory_bank_creator_array`?

Reply:
I remember searching this API when I wrote the code and didn't find it. Turns out it is named differently:  `memory_bank_array_creator`